apps/shell/app.js

```diff
@@ -72,15 +72,24 @@ glob.sync(path.join((process.env.OOD_CLUSTERS || '/etc/ood/config/clusters.d'),
     let isDefault = config.v2.login.default;
     host_whitelist.add(host);
     if (isDefault) default_sshhost = host;
+    if (isFirst) {
+      first_available_host = host;
+      isFirst = false;
```

Comment:
You don't need an extra boolean, just do:

    if(! first_available_host) first_available_host = host;